widget/table.go

```diff
@@ -90,41 +79,51 @@ func (t *Table) CreateRenderer() fyne.WidgetRenderer {
 	return r
 }
 
-// Selection will mark the specified cell (at row, col) to be marked as selected.
-func (t *Table) Selection() (row, col int) {
-	return t.selectedRow, t.selectedColumn
+// Select will mark the specified cell as selected.
+func (t *Table) Select(id TableCellID) {
```

Comment:
The is a small optimization you could add here to check if id == selectedCell

Reply:
Resolved

Reply:
It would be safer to just write `t.selectedCell = &id` - no need to dereference any data and no need to nil check

Reply:
Resolved

Reply:
I think I misread this first time round - I thought it was assigning.
As a comparison I might have been wrong... This equality check will only work when the pointers are the same - and given it's an exposed API this will not always be the case.
Sorry!